cmd/bnscli/cmd_multisig.go

```diff
@@ -0,0 +1,136 @@
+		adminThresholdFl      = fl.Uint("admin", 0, "Admin threshold value. Must be greater than 0.")
+	)
+	fl.Parse(args)
+
+	if *activationThresholdFl == 0 {
+		flagDie("activation threshold cannot be zero")
+	}
+	if *adminThresholdFl == 0 {
+		flagDie("admin threshold cannot be zero")
+	}
+
+	var tx app.Tx
+
+	if len(*updateFl) != 0 {
+		tx = app.Tx{
+			Sum: &app.Tx_UpdateContractMsg{
+				UpdateContractMsg: &multisig.UpdateContractMsg{
+					Metadata:            &weave.Metadata{Schema: 1},
+					ContractID:          *updateFl,
+					ActivationThreshold: multisig.Weight(*activationThresholdFl),
+					AdminThreshold:      multisig.Weight(*adminThresholdFl),
+				},
+			},
+		}
+	} else {
+		tx = app.Tx{
+			Sum: &app.Tx_CreateContractMsg{
+				CreateContractMsg: &multisig.CreateContractMsg{
+					Metadata:            &weave.Metadata{Schema: 1},
+					ActivationThreshold: multisig.Weight(*activationThresholdFl),
+					AdminThreshold:      multisig.Weight(*adminThresholdFl),
+				},
+			},
+		}
+	}
+
+	_, err := writeTx(output, &tx)
+	return err
+}
+
+func cmdWithMultisigParticipant(input io.Reader, output io.Writer, args []string) error {
+	fl := flag.NewFlagSet("", flag.ExitOnError)
+	fl.Usage = func() {
+		fmt.Fprintln(flag.CommandLine.Output(), `
+Read a multisig create or update transaction from the input and modify it by
+adding a participant. Returned transaction is the original content with a
+participant added.
+		`)
+		fl.PrintDefaults()
+	}
+	var (
+		sigFl    = flAddress(fl, "sig", "", "Participant signature/address.")
+		weightFl = fl.Uint("weight", 1, "Participant weight.")
+	)
+	fl.Parse(args)
+
+	tx, _, err := readTx(input)
+	if err != nil {
+		return fmt.Errorf("cannot read input transaction: %s", err)
+	}
+
+	msg, err := tx.GetMsg()
+	if err != nil {
+		return fmt.Errorf("cannot extract transaction message: %s", err)
+	}
+
+	switch msg := msg.(type) {
+	case *multisig.CreateContractMsg:
+		msg.Participants = append(msg.Participants, &multisig.Participant{
+			Signature: *sigFl,
+			Weight:    multisig.Weight(*weightFl),
+		})
+	case *multisig.UpdateContractMsg:
+		msg.Participants = append(msg.Participants, &multisig.Participant{
+			Signature: *sigFl,
+			Weight:    multisig.Weight(*weightFl),
+		})
+	default:
+		return fmt.Errorf("message %T cannot be modified to contain multisig participant", msg)
+	}
+
+	_, err = writeTx(output, tx)
+	return nil
+}
+
+func cmdWithMultisig(input io.Reader, output io.Writer, args []string) error {
+	fl := flag.NewFlagSet("", flag.ExitOnError)
+	fl.Usage = func() {
+		fmt.Fprintln(flag.CommandLine.Output(), `
+Read a transaction from the input and attach any number of provided multisig
+contract IDs to that transaction.
+		`)
+		fl.PrintDefaults()
+	}
+	fl.Parse(args)
+
+	tx, _, err := readTx(input)
+	if err != nil {
+		return fmt.Errorf("cannot read input transaction: %s", err)
+	}
+
+	for _, mid := range fl.Args() {
+		tx.Multisig = append(tx.Multisig, []byte(mid))
```

Comment:
This is not cool.... You just take ascii strings and turn into bytes?

Earlier discussions were always about 
* int -> turned into 8-byte bigendian sequence
* hex -> parse to bytes, user defines the length
depending on which made more sense for the command.

turning ascii into bytes will almost never be able to get the data point we want.

Reply:
Thank for having a closer look. It seems as I often mix bytes/address/sequence/condition.

I have addressed this problem in https://github.com/iov-one/weave/pull/759/commits/f825de675756e8a16fdeb45f893f2848a4555758 by adding a `unpackSequence` and `flSeq` functions. Please have a look again.

Reply:
:pray: once more please.

While using the command, I have realized that there must be an explicit format information. Issue is addressed in https://github.com/iov-one/weave/pull/759/commits/18076ce9f925ab98e4663e5f47bf7ee5c89c4f9e

> When parsing a user provided sequence value, require explicit format
inforamtion.
Before the `unpackSequence` function was trying to guess the format. The
problem is that for example `1234` is a both valid decimal, hex and
base64 representation of a value. This could lead to false positives
when parsing.
> 
> With this change, always default to the decimal parser. Allow to provide
a prefix (ie `hex:` or `base64:`) to provide a value using non default
encoding.